matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/queue/QueueMemento.kt

```diff
@@ -107,10 +109,18 @@ internal class QueueMemento @Inject constructor(
                                 info.redactionLocalEcho?.let { localEchoRepository.getUpToDateEcho(it) }?.let {
                                     localEchoRepository.updateSendState(it.eventId!!, it.roomId, SendState.UNSENT)
                                     // try to get reason
-                                    val reason = it.content?.get("reason") as? String
+                                    val body = it.content.toModel<EventRedactBody>()
                                     if (it.redacts != null && it.roomId != null) {
                                         Timber.d("## Send -Reschedule redact $info")
-                                        eventProcessor.postTask(queuedTaskFactory.createRedactTask(it.eventId, it.redacts, it.roomId, reason))
+                                        eventProcessor.postTask(
+                                                queuedTaskFactory.createRedactTask(
+                                                        redactionLocalEcho = it.eventId,
+                                                        eventId = it.redacts,
+                                                        roomId = it.roomId,
+                                                        reason = body?.reason,
+                                                        withRelations = body?.withRelations ?: body?.withRelationsUnstable
```

Comment:
if we want to have both stable and unstable fields, we can introduce a helper method in `EventRedactBody` class as:

`fun getBestWithRelations() = withRelations ?: withRelationsUnstable`

Reply:
I removed the stable field: https://github.com/vector-im/element-android/pull/7988/commits/0870cec118152665f95741f70680e885dae33792